packages/@ama-terasu/cli/src/helpers/module.helper.ts

```diff
@@ -0,0 +1,151 @@
+import * as chalk from 'chalk';
+import { exec } from 'node:child_process';
+import { existsSync, promises as fs } from 'node:fs';
+import { dirname, join, resolve } from 'node:path';
+import { promisify } from 'node:util';
+import type { PackageJson } from 'type-fest';
+
+/** Package information resulting of a NPM search command */
+export type SearchResult = Pick<PackageJson, 'name' | 'description' | 'keywords' | 'maintainers' | 'version'> & { scope: string; date: Date; links: Record<string, string> };
+
+/** Module discovered */
+export type ModuleDiscovery = (PackageJson | SearchResult) & { name: string; moduleName: string; resolutionPath: string | null };
```

Comment:
I tried first but it was quite complicated because `PackageJson` is a type that is coming is constraints and build with composition as well.
I can move out the documented fields to a dedicated interface tp make it more easily readable

Reply:
In that case yes having the additional fields properly documented in an interface can be nice